components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java

```diff
@@ -657,6 +656,30 @@ private void sendToFrameworkForLogout(HttpServletRequest request, HttpServletRes
         sendRequestToFramework(request, response, sessionDataKey, FrameworkConstants.RequestType.CLAIM_TYPE_SAML_SSO);
     }
 
+    /**
+     * Send the Artifact as a response to a SAML authentication request.
+     *
+     * @param resp                 Response object to be sent.
+     * @param relayState           Relay state of the request.
+     * @param artifact             Generated SAML2 artifact.
+     * @param assertionConsumerUrl ACU to be sent.
+     * @throws IOException
+     */
+    private void sendArtifact(HttpServletResponse resp, String relayState, String artifact,
+                              String assertionConsumerUrl) throws IOException {
+
+        // Set the HTTP Headers: HTTP proxies and user agents should not cache the artifact
+        resp.addHeader("Pragma", "no-cache");
+        resp.addHeader("Cache-Control", "no-cache");
+        String encodedArtifact = URLEncoder.encode(artifact, "UTF-8");
+
+        String queryParams = "?" + SAMLSSOConstants.SAML_ART + "=" + encodedArtifact;
+        if(relayState != null) {
+            queryParams += "&" + SAMLSSOConstants.RELAY_STATE + "=" + relayState;
```

Comment:
Shall we use FrameworkUtils.appendQueryParamsStringToUrl(url, queryParams) to append query params to url?